File: frontend/src/app/model/purchase-request.ts
import { Address } from "./address";
import { Client } from "./client";
import { Item } from "./item";
import { RequestOrder } from "./request-order";

export class PurchaseRequest {
    client!:Client;
    requestOrder! :RequestOrder;
    fromAddress! :Address;
    toAddress!: Address;
    items! : Item[];
}

File: backend/restaurant-system/src/main/java/com/restaurant/restaurantsystem/dto/PurchaseRequest.java
package com.restaurant.restaurantsystem.dto;

import com.restaurant.restaurantsystem.entity.Address;
import com.restaurant.restaurantsystem.entity.Client;
import com.restaurant.restaurantsystem.entity.Item;
import com.restaurant.restaurantsystem.entity.RequestOrder;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;


@Data
@AllArgsConstructor
@NoArgsConstructor
public class PurchaseRequest {
    private Client client;
    private RequestOrder requestOrder;
    private List<Item> items = new ArrayList<>();
    private Address fromAddress;
    private Address toAddress;

}
